src/e2e/java/teammates/e2e/cases/e2e/StudentFeedbackResultsPageE2ETest.java

```diff
@@ -0,0 +1,278 @@
+import teammates.common.util.Const;
+import teammates.e2e.pageobjects.AppPage;
+import teammates.e2e.pageobjects.StudentFeedbackResultsPage;
+
+/**
+ * SUT: {@link Const.WebPageURIs#STUDENT_SESSION_RESULTS_PAGE}.
+ */
+public class StudentFeedbackResultsPageE2ETest extends BaseE2ETestCase {
+    private StudentFeedbackResultsPage resultsPage;
+    private FeedbackSessionAttributes openSession;
+
+    @Override
+    protected void prepareTestData() {
+        testData = loadDataBundle("/StudentFeedbackResultsPageE2ETest.json");
+        removeAndRestoreDataBundle(testData);
+
+        openSession = testData.feedbackSessions.get("Open Session");
+    }
+
+    @Test
+    public void testAll() {
+
+        ______TS("unregistered student can access results");
+        StudentAttributes unregistered = testData.students.get("Unregistered");
+        AppUrl url = createUrl(Const.WebPageURIs.SESSION_RESULTS_PAGE)
+                .withCourseId(unregistered.course)
+                .withStudentEmail(unregistered.email)
+                .withSessionName(openSession.getFeedbackSessionName())
+                .withRegistrationKey(getKeyForStudent(unregistered));
+        logout();
+        resultsPage = AppPage.getNewPageInstance(browser, url, StudentFeedbackResultsPage.class);
+
+        resultsPage.verifyFeedbackSessionDetails(testData.feedbackSessions.get("Open Session"));
+
+        ______TS("registered student can access results");
+        StudentAttributes student = testData.students.get("Alice");
+        url = createUrl(Const.WebPageURIs.STUDENT_SESSION_RESULTS_PAGE)
+                .withUserId(student.googleId)
+                .withCourseId(openSession.getCourseId())
+                .withSessionName(openSession.getFeedbackSessionName());
+        resultsPage = loginAdminToPage(url, StudentFeedbackResultsPage.class);
+
+        resultsPage.verifyFeedbackSessionDetails(testData.feedbackSessions.get("Open Session"));
+
+        ______TS("questions loaded");
+        for (int i = 1; i <= 11; i++) {
+            resultsPage.verifyQuestionDetails(i, testData.feedbackQuestions.get("qn" + i));
+        }
+
+        ______TS("questions with no responses not shown");
+        resultsPage.verifyQuestionNotPresent(12);
+
+        ______TS("verify responses");
+        // qn11 is a contribution question so we only need to check the statistics for that question
+        for (int i = 1; i <= 10; i++) {
+            verifyResponseDetails(student, testData.feedbackQuestions.get("qn" + i));
+        }
+
+        ______TS("verify statistics - numscale");
+        resultsPage.verifyNumScaleStatistics(5, student, getReceivedResponses(student,
+                testData.feedbackQuestions.get("qn5")));
+
+        ______TS("verify statistics - rubric");
+        FeedbackQuestionAttributes rubricsQn = testData.feedbackQuestions.get("qn10");
+        resultsPage.verifyRubricStatistics(10, rubricsQn, getReceivedResponses(student, rubricsQn),
```

Comment:
Let's just hard-code the statistics here. Similarly for the numscale question statistics.